Formula/lmod.rb

```diff
@@ -0,0 +1,64 @@
+class Lmod < Formula
+  desc "Lua-based environment modules system to modify PATH variable"
+  homepage "https://www.tacc.utexas.edu/research-development/tacc-projects/lmod"
+  url "https://github.com/TACC/Lmod/archive/7.7.16.tar.gz"
+  sha256 "cf14183f9954bf0d4c22db4b7db3a6459c8cb607b9d1ff25e18e0989b48bf58b"
+
+  depends_on "lua"
+
+  resource "luafilesystem" do
+    url "https://github.com/keplerproject/luafilesystem/archive/v1_7_0_2.tar.gz"
+    sha256 "23b4883aeb4fb90b2d0f338659f33a631f9df7a7e67c54115775a77d4ac3cc59"
+  end
+
+  resource "luaposix" do
+    # Don't use v34.0.2, it's broken (at least on macOS)
+    url "https://github.com/luaposix/luaposix/archive/v34.0.1.tar.gz"
+    sha256 "43e7fb1fa4a8b905dd256703aa62116937be72d07d3a46b66a8e9b2264921129"
+  end
+
+  def install
+    luapath = libexec/"vendor"
+    ENV["LUA_PATH"] = "#{luapath}/share/lua/5.3/?.lua" \
+                      ";#{luapath}/share/lua/5.3/?/init.lua"
+    ENV["LUA_CPATH"] = "#{luapath}/lib/lua/5.3/?.so"
+
+    resources.each do |r|
+      r.stage do
+        system "luarocks", "make", "--tree=#{luapath}"
+      end
+    end
+
+    system "./configure",
+      "--disable-debug",
+      "--disable-dependency-tracking",
+      "--disable-silent-rules",
+      "--prefix=#{prefix}"
+    system "make", "install"
+
+    # Move prefix/lmod/VERSION/lmod/VERSION/* to prefix/lmod/VERSION/*
+    mv Dir[prefix/"lmod/#{version}/*"], prefix
+    rmdir prefix/"lmod/#{version}"
+    (prefix/"lmod").install_symlink ".." => version
+    Dir[prefix/"lmod/init/*"].reject { |f| f["lmodrc.lua"] }.each do |f|
+      inreplace f, "lmod/lmod", "lmod"
+    end
+  end
+
+  def caveats
+    s = <<~EOS
+      To use LMod, you should add the init script to the shell you are using.
+
+      For example, the bash setup script is here: $(brew --prefix lmod)/lmod/init/profile and you can source it in your bash setup or link to it.
+
+      If you use fish, use $(brew --prefix lmod)/lmod/init/fish, such as:
+        ln -s #{prefix}/lmod/init/fish ~/.config/fish/conf.d/00_lmod.fish
+    EOS
+
+    s
+  end
+
+  test do
+    system "#{prefix}/lmod/init/sh"
```

Comment:
Can we do something more than run the init script?

Reply:
Running the `ml` command catches most Lua issues, and is actually what I usually check first, so I'll add that.

Reply:
The ml command requires the init script to be sourced, not sure how to do that through system.